### PR TITLE
Fix k0s failure due to partial write of embedded binaries.

### LINKED ISSUE #1989: K0s does not recover if embedded binaries are written only partially to a disk
### Before creating an issue, make sure you've checked the following:

- [X] You are running the latest released version of k0s
- [X] Make sure you've searched for existing issues, both open and closed
- [X] Make sure you've searched for PRs too, a fix might've been merged already
- [X] You're looking at docs for the released version, "main" branch docs are usually ahead of released versions.

### Version

v1.23.8+k0s.0

### Platform

```shell
Distributor ID:	Debian
Description:	Debian GNU/Linux 11 (bullseye)
Release:	11
Codename:	bullseye
```

### What happened?

k0s wrote empty (or partial) binary files to disk which blocks k0s from starting properly (if for example kubelet binary is broken). Restart does not help because these broken binaries do have correct timestamp. Only way to recover is manual removal of broken binaries and then k0s restart.

### Steps to reproduce

1.
2.
3.

### Expected behavior

_No response_

### Actual behavior

_No response_

### Screenshots and logs

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by makhov**
Signed-off-by: Alexey Makhov <amakhov@mirantis.com>

## Description

Fixes #1989

Writes the original size of the embedded files to the static assets and unpacks files if the size doesn't match.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [x] Manual test
- [ ] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,7 +14,7 @@ else
   endif
   HOST_ARCH := amd64
 endif
-undefine HOST_HARDWARE
+unexport HOST_HARDWARE
```
Comment (twz123) [inline Makefile:17]: That's interesting. What's the reason for this?

File: pkg/assets/stage.go [hunk 2]
```diff
@@ -29,11 +29,11 @@ import (
 
 // EmbeddedBinaryNeedsUpdate returns true if the provided embedded binary file should
 // be updated. This determination is based on the modification times of both the provided
-// executable, and embedded binary executable. It is expected that the embedded binary
+// executable, and embedded binary executable, and on the binary size. It is expected that the embedded binary
```
Comment (twz123) [inline pkg/assets/stage.go:32]: Maybe this:

> This determination is based on the modification times and file sizes of both the provided executable and the embedded executable.

**Review by twz123 — APPROVED** [review]
Would be cool if we'd have some tests for this.

File: Makefile (hunk 1 above)
Comment (makhov) [reply Makefile]: I had an old version of the GNU make and it didn't work. Changed it back

File: pkg/assets/stage.go (hunk 2 above)
Comment (makhov) [reply pkg/assets/stage.go]: Done

File: Makefile (hunk 1 above)
Comment (twz123) [reply Makefile]: Ah, good to know. Technically, the else/undefine block isn't even necessary. So if that remains a problem we could also just remove it completely.

**Review by twz123 — APPROVED** [review]

**Pull request merged**